### ISSUE TITLE
query's input stream doesn't trigger the query callback

### ISSUE THREAD

**Issue by mtt-artis** [timdeschryver/rx-query#35]
I have a project with angular 14 and I'd like to add rx-query.
I have several components on the same page whose data is retrieved from an API.

`
Query$ = query('query',
    this.activatedRoute.queryParams.pipe(map(({id, currency}) => ({id, currency}))),
    ({id, currency}) => this.repo.getData(id, currency)
  )
`

Unfortunately, the query's input stream doesn't trigger the query callback.
I fixed it by changing `asapScheduler` to `asyncScheduler`. but I don't know enough rxjs to understand the origin of the problem.

Do you expect a side effect from that?
If not, can I create a PR?

**Comment by SchlammSpringer** [comment]
thxs for the hint, tried it and it works, so for me the same question is, what are the side effects? 
Or could it be also a problem because of this Bug:

https://github.com/ReactiveX/rxjs/issues/6747

**Comment by mtt-artis** [comment]
this might be the problem yes,

I made a [stackblitz](https://stackblitz.com/edit/rx-query-with-rxjs7?file=package.json) where you can see the problem starts with rxjs 7.4 (7.3 is fine).

I'm on a project with more than 40 queries and rx-query behaves as i want with the `asyncScheduler`.

**Comment by SchlammSpringer** [comment]
for us the downgrade to rx.js 7.4 is working, with a little jest config editing.
https://github.com/nrwl/nx/issues/7844#issuecomment-1139427677

Still the best solution would be.

- pull request to asyncScheduler / with new peer dependencies
- wait until the bug is fixed and then only pull request with new peer depencies

@timdeschryver could you give us a hint which solution/pull request we should focus on.

**Comment by timdeschryver** [comment]
@SchlammSpringer I prefer the first option, but I suspect some tests will fail (If I remember correctly). Feel free to create a PR.
I was also thinking to make the scheduler configurable, which could be useful in both versions.

**Comment by mtt-artis** [comment]
The asyncScheduler does not guarantee the order of events. 
This can be seen particularly on the "can refresh on demand" test where the result is messed up : 
```ts
// at rx-query/__tests__/query.test.ts:501
it.only('can refresh on demand', async () => {
    const values = [];
    let i = 20;

    for await (const value of eachValueFrom(
        query('test', () => of(i++)).pipe(takeWhile(() => i <= 24, true)),
    )) {
        refreshQuery('test');
        values.push(value);
    }
    console.log(valuesWithoutMutate(values))
    // ...
})
// [
//   { status: 'loading', retries: 0 },
//   { status: 'success', data: 20 },
//   { status: 'refreshing', data: 20, retries: 0 },
//   { status: 'refreshing', data: 20, retries: 0 },
//   { status: 'success', data: 21 },
//   { status: 'success', data: 22 },
//   { status: 'refreshing', data: 21, retries: 0 },
//   { status: 'refreshing', data: 22, retries: 0 },
//   { status: 'refreshing', data: 22, retries: 0 },
//   { status: 'success', data: 23 },
//   { status: 'success', data: 24 }
// ]
```
This behavior breaks several tests...

**Comment by SchlammSpringer** [comment]
Same result here. 
But is the execution order really important? 
In the tests there is also an isolation problem with the
`resetQueryCache`. 
Single test execution behaves different than suite execution.
First impression from us, change to `asyncScheduler` is not a viable quickfix, or we don´t understand the cache implementation good enough.

**Comment by timdeschryver** [comment]
Oh shoot.. I accidently pushed some changes to `master` 😅
I canceled the workflow to prevent a release.
Could you try the version on `master` please?

**Comment by SchlammSpringer** [comment]
@timdeschryver thanks for the push. First quick impression, still strange behavior. We will investigate further.

**Comment by timdeschryver** [comment]
Thanks @SchlammSpringer .
What about https://github.com/timdeschryver/rx-query/pull/37 ? 
It seems to be okay here, but I still want the tests to pass 😅

**Comment by SchlammSpringer** [comment]
@timdeschryver thxs for the PR => https://github.com/timdeschryver/rx-query/pull/37#issuecomment-1236079013

**Comment by timdeschryver** [comment]
Should be resolved in v2.0.1
If that's not the case feel free to open a new issue.
